tools/opt/CMakeLists.txt

```diff
@@ -37,11 +37,12 @@ add_llvm_tool(opt
   opt.cpp
 )
 
+if(WIN32)
 target_link_libraries(opt
    DxilConvPasses
 )
-
 add_dependencies(opt DxilConvPasses)
+endif(WIN32)
```

Comment:
This isn't quite what I expected to resolve the Linux build issue, but it seems to work with a small addition to lib/HLSL/LLVMBuild.txt. Add `DXIL` to `required_libraries`. 

The reason for this and other dependency additions to make Linux build properly is that Linux is more persnickety about the order in which link dependencies are listed. The Windows and even MacOS linkers will silently reorganize your objects when necessary to resolve dependencies. Linux either doesn't think you deserve that courtesy or else feels the rare case where you have multiple objects that export the same symbols and might get the wrong one if your order is not respected is worth causing this many headaches. If dependencies are properly identified through the build system, cmake will form the necessary ordering so that all is well. The dependencies are currently not all properly identified, but this brings us one step closer!